src/sources/prometheus/remote_write.rs

```diff
@@ -340,13 +340,22 @@ mod test {
 
 #[cfg(all(test, feature = "prometheus-integration-tests"))]
 mod integration_tests {
+    use std::net::{SocketAddr, ToSocketAddrs as _};
     use tokio::time::Duration;
 
     use super::*;
     use crate::test_util::components::{run_and_assert_source_compliance, HTTP_PUSH_SOURCE_TAGS};
 
-    fn source_receive_address() -> String {
-        std::env::var("SOURCE_RECEIVE_ADDRESS").unwrap_or_else(|_| "127.0.0.1:9102".into())
+    fn source_receive_address() -> SocketAddr {
+        let address =
+            std::env::var("SOURCE_RECEIVE_ADDRESS").unwrap_or_else(|_| "runner:9102".into());
```

Comment:
Should this still be else'd into localhost for people trying to run this outside of docker... is that something we want to support still?

Reply:
🤷🏻 I'm open to it. How are you thinking to make that work?

Reply:
I'd expect `SOURCE_RECEIVE_ADDRESS` to have the docker dns name of `runner` and if not present assume the user is just on localhost - there's not much support on the "services" side and it'd be very choose your own adventure.